wled00/wled_math.cpp

```diff
@@ -12,6 +12,12 @@
 
 #define modd(x, y) ((x) - (int)((x) / (y)) * (y))
 
+// Note: cos_t, sin_t and tan_t are very accurate but may be slow
+// the math.h functions use several kB of flash and are to be avoided if possible
+// sin16_t / cos16_t are faster and much more accurate than the fastled variants
+// sin_approx and cos_approx are float wrappers for sin16_t/cos16_t and have an accuracy of +/-0.0015 compared to sinf()
+// sin8_t / cos8_t are fastled replacements and use sin16_t / cos16_t. Slightly slower than fastled version but very accurate
```

Comment:
👍 very good that we have a replacement for sin8 and cos8. 
These FastLed functions were so inaccurate that you could not even draw a circle with them.

Reply:
Here is the comparison between the new version and fastled (compared to `sinf()` )
sin8():
![image](https://github.com/user-attachments/assets/1ee22e19-613a-4c82-a8e6-602847274a69)

sin16():
![image](https://github.com/user-attachments/assets/05c6ca5d-b6ba-46c9-ab06-60694278950d)

the new sin8_t() is a bit slower than the fastled variant due to overhad in converstion to 16bit range. The approximation can be re-written to any scale so it could be done for 8bit inputs/outputs natively but since sin16_t() is now faster than sin8() it would make more sense to use the sin16_t() instead wherever speed is of utmost importance.

Reply:
Thanks, I like the graphs 👍 very useful to understand accuracy and limitations.

Reply:
> but since sin16_t() is now faster than sin8() it would make more sense to use the sin16_t() instead wherever speed is of utmost importance.

I agree - I don't think we need a specialized sin8, as most "sine wave" effects are using beatsin88() and not sin8.

Reply:
Another idea about sin8_t and cos8_t:

You could move these two functions into fx.cpp and make them "static inline". This will allow the compiler to optimize out function calls (->faster). I think there is not other place where these 8bit version would be needed.

Reply:
> I think there is not other place where these 8bit version would be needed.

It is used in beatsin8, which is also used in sound simulation in util.cpp.
While making that an inline function could boost speed, I don't think it those extra few instructions will do any harm. If we really want it faster, it should have its own dedicated function (which is doable, it uses roughly 100 bytes)

Reply:
@DedeHai can you test this variant? 

```c++
uint8_t sin8_t(uint8_t theta) {
  int32_t sin16 = sin16_t((uint16_t)theta * 257); // 255 * 257 = 0xFFFF
  sin16 += 0x7FFF + 128; //shift result to range 0-0xFFFF, +128 for rounding
  return min(sin16, int32_t(0xFFFF)) >> 8; // min performs saturation, and prevents overflow
}
```

 It just adds one or two CPU instructions for "max", and should keep the error below +-0.5

Reply:
that works, maybe a good tradoff between accuracy and speed.
![image](https://github.com/user-attachments/assets/9bc69995-6f02-46cb-ab45-c25b458d2a2b)

Reply:
👍 👍

Reply:
> that works, maybe a good tradoff between accuracy and speed. 

I've just compared your new trig functions against the old one, using the "Lissajous" effect on 128x64. It's really a major difference. The "fastled" lissajous curve looks very edgy and somehow _wrong_, more like a twisted hexagon. With your new functions, I see a perfect wave.

 Good job👍 thanks.